File: Leetcoding-Actions/Explore-Learn/Arrays/checkIfNAndItsDoubleExist.java
class Solution {
    public boolean checkIfExist(int[] arr) {
        /**
        * T: O(n) and S: O(n)
        */
        Set<Integer> set = new HashSet<>();
        for (int i = 0; i < arr.length; i++) {
            if (set.contains(2 * arr[i])) {
                return true;
            } else {
                set.add(arr[i]);
            }
        }
        Set<Integer> revSet = new HashSet<>();
        for (int i = arr.length - 1; i >= 0; i--) {
            if (revSet.contains(2 * arr[i])) {
                return true;
            } else {
                revSet.add(arr[i]);
            }
        }
        return false;
    }
}

File: Leetcoding-Actions/Explore-Learn/Arrays/checkIfNAndItsDoubleExist.py
class Solution:
    def checkIfExist(self, arr: List[int]) -> bool:
        '''
        T: O(n) and S: O(n)
        '''
        numSet = set()
        for i in range(len(arr)):
            if 2 * arr[i] in numSet:
                return True
            else:
                numSet.add(arr[i])
        
        revNumSet = set()
        for i in range(len(arr)-1, -1, -1):
            if 2 * arr[i] in revNumSet:
                return True
            else:
                revNumSet.add(arr[i])
                
        return False
